pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java

```diff
@@ -730,6 +730,25 @@ public void testUpdateDynamicLocalConfiguration() throws Exception {
         });
     }
 
+    @Test(timeOut = 30000)
+    public void testUpdateCustomDynamicConfiguration() throws Exception {
+        final var key = "custom-key";
+        final var invalidValue = "invalid-value";
+        pulsar.getBrokerService().registerCustomDynamicConfiguration(key, value -> !value.equals(invalidValue));
+        assertThrows(IllegalArgumentException.class, () -> pulsar.getBrokerService()
+                .registerCustomDynamicConfiguration("brokerShutdownTimeoutMs", __ -> true));
+
+        admin.brokers().updateDynamicConfiguration(key, "valid-value");
+
+        Awaitility.waitAtMost(10, TimeUnit.SECONDS).untilAsserted(() -> {
```

Comment:
Why need `Awaitility` here? I see the `updateDynamicConfiguration` has sync updated to storage.

Reply:
It's right that we don't need `Awaitility` here. I copied the code from `testUpdateDynamicLocalConfiguration` before. But I just found they are not the same case because `testUpdateDynamicLocalConfiguration` checks if the configuration field has been updated, which needs the data synchronization in `handleDynamicConfigurationUpdates`.